velox/core/PlanNode.cpp

```diff
@@ -846,35 +846,90 @@ PlanNodePtr MergeJoinNode::create(const folly::dynamic& obj, void* context) {
       outputType);
 }
 
-CrossJoinNode::CrossJoinNode(
+NestedLoopJoinNode::NestedLoopJoinNode(
     const PlanNodeId& id,
+    JoinType joinType,
+    TypedExprPtr joinCondition,
     PlanNodePtr left,
     PlanNodePtr right,
-    RowTypePtr outputType)
+    const RowTypePtr& outputType)
     : PlanNode(id),
+      joinType_(joinType),
+      joinCondition_(std::move(joinCondition)),
       sources_({std::move(left), std::move(right)}),
-      outputType_(std::move(outputType)) {}
+      outputType_(outputType) {
+  VELOX_CHECK(
+      core::isInnerJoin(joinType_) || core::isLeftJoin(joinType_) ||
+          core::isRightJoin(joinType_) || core::isFullJoin(joinType_),
+      "Currently NestedLoopJoin only supports inner and outer join");
```

Comment:
drop "Currently" and include 'joinType' in the error message for easier troubleshooting:

```
NestedLoopJoin only supports inner, left, right and full joins: {}
```